api/transport/request.go

```diff
@@ -62,6 +62,10 @@ type Request struct {
 
 	// Request payload.
 	Body io.Reader
+
+	// TransportHeaders are other transport headers that the transport may expose
+	// to the encodings.
+	TransportHeaders map[string]string
```

Comment:
Yes, I did see the description, and this is my recommendation. See https://github.com/yarpc/yarpc-go/blob/master/encoding/x/protobuf/types.go#L45.